Enable many ruff checks

inknos opened the pull request:
Depends on https://github.com/containers/podman-py/pull/473

inknos commented:
```
podman/domain/images_manager.py:291: in ImagesManager
    body: list[dict[str, Any]],  # noqa: A003
E   TypeError: 'function' object is not subscriptable
```

The errors come from removing the deprecated typing imports for Dict, List, Tuple and Type. Specifically, list is used in place of List and it's shadowed by the `list` method. Need to think of how to go around the issue without breaking the api in the first place

jwhonce commented on podman/domain/images_manager.py:
```diff
@@ -31,6 +32,8 @@
 logger = logging.getLogger("podman.images")
 
+builtin_list = list
+
```
From my reading:

```
import builtins
>>> builtins.list
```

is more explicit about what is happening in this code. We are stuck with the naming collision. I think the safety of the typing is more important than the code bloat this will introduce. Or we can stay with List[].

inknos replied:
`typing.List` is deprecated, that's why we should move away from it if we can.

https://docs.python.org/3/library/typing.html#typing.List

I like your solution better though, it's more clear. I believe the `list` shadowing will not be problematic for users though, but the ambiguity in returns would eventually flag unexpected errors in IDEs, testing, checkers and such.

inknos replied:
@jwhonce I updated the PR with your suggestion

jwhonce replied:
@inknos The joys of trying to graph type checking on a language that is not statically typed.

inknos replied:
I know...

jwhonce commented on podman/domain/images_manager.py:
```diff
@@ -200,8 +202,8 @@ def prune(
 
-        deleted: List[Dict[str, str]] = []
-        error: List[str] = []
+        deleted: list[dict[str, str]] = []
+        error: list[str] = []
```
```suggestion
        deleted: builtins.list[dict[str, str]] = []
        error: builtins.list[str] = []
```
May be more correct. I did not research where in the scope this list() is anchored.

inknos replied:
I changed all the ambiguous for the linter, probably, for the human eye is better to be more explicit with the checks as you suggest. I'll change it this way

jwhonce commented on podman/domain/images_manager.py:
```diff
@@ -409,7 +415,7 @@ def pull(
             obj = json.loads(item)
             if all_tags and "images" in obj:
-                images: List[Image] = []
+                images: list[Image] = []
```
```suggestion
                images: builtins.list[Image] = []
```
See above

jwhonce commented on podman/domain/images_manager.py:
```diff
@@ -473,22 +479,22 @@ def remove(
 
         body = response.json()
-        results: List[Dict[str, Union[int, str]]] = []
+        results: list[dict[str, Union[int, str]]] = []
```
```suggestion
        results: builtins.list[dict[str, Union[int, str]]] = []
```

jwhonce commented on podman/domain/manager.py:
```diff
@@ -2,15 +2,16 @@
 
 
-class PodmanResource(ABC):
+class PodmanResource:
```
Why remove ABC base clase?

inknos replied:
I think I might have been too strict with this one, but my reasoning was the following. The class has no @abstract method so it's not technically an interface. Now that I think about it, we should keep it abstract to prevent the instantiation of the class. I will bring back the ABC inheritance and silence the error on that line

jwhonce commented on podman/domain/networks_manager.py:
```diff
@@ -161,8 +161,8 @@ def list(self, **kwargs) -> List[Network]:
-        self, filters: Optional[Dict[str, Any]] = None
-    ) -> Dict[api.Literal["NetworksDeleted", "SpaceReclaimed"], Any]:
+        self, filters: Optional[dict[str, Any]] = None
+    ) -> dict[api.Literal["NetworksDeleted", "SpaceReclaimed"], Any]:
```
```suggestion
    ) -> dict[Literal["NetworksDeleted", "SpaceReclaimed"], Any]:
```
Do we need to use api.Literal any more?

inknos replied:
probably not, I'll handle it in a separate commit

jwhonce reviewed (changes requested):
Some nits and some questions. Looking good!

inknos commented:
@jwhonce I addressed the `builtins.list` comments. now the calls are explicit within the classes with the shadowed `list` methods. Then, I removed `api.Literal` and brought back `ABC` abstract subclassing.

Also, I removed the last commit as it was more like a placeholder for future ruff checks and I don't like it upstream but rather track it in a different way.

inknos commented:
@jwhonce all should be good now

jwhonce reviewed (approved):
Looks good.

LGTM

baude commented:
/lgtm
